### PR TITLE
VACMS-10643: php 8.1 related issues

### LINKED ISSUE #10643: php 8.1 related issues
`PHP Parse error:  syntax error, unexpected token "match" in /var/www/html/docroot/vendor/caxy/php-htmldiff/lib/Caxy/HtmlDiff/HtmlDiff.php on line 699`

Update drush to v11.

drupal/styleguide `Deprecated function: strpos(): Passing null to parameter #1`

`Deprecated function: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated in Drupal\node_link_report\Plugin\Block\NodeLinkReport->isAllowed() (line 895 of modules/contrib/node_link_report/src/Plugin/Block/NodeLinkReport.php).`

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #10643

## Testing done

Tested locally on ddev using php 8.1

## QA steps

You will need to update you ddev config to php 8.1 to test locally. There will probably be deprecations that I missed. If you find any please note them and I will address them. 
I will also be checking that everything is working correctly on Tugboat with php 7.4, if you could please do the same. 

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -263,7 +262,7 @@
         "sort-packages": true,
         "optimize-autoloader": true,
         "platform": {
-            "php": "7.4"
+            "php": "8.1"
```
Comment (ndouglas) [inline composer.json:265]: I was curious about whether this might cause problems.

[Apparently not](https://getcomposer.org/doc/06-config.md#platform).

**Review by ndouglas — COMMENTED** [review]
This looks great!  I'm holding off on approval until the tests pass, because some version changes or other deployment changes could conceivably cause large issues.  But awesome work!  👍🏻

File: composer.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
         "drupal/ctools_block": "^3.2",
         "drupal/danse": "^2.1",
         "drupal/danse_content_moderation": "^2.0",
-        "drupal/diff": "1.0",
+        "drupal/diff": "1.x-dev@dev",
```
Comment (swirtSJW) [inline composer.json:51]: I prefer to not track dev if we can avoid it. Can we just include the patch that makes the switch to "caxy/php-htmldiff"

File: composer.json [hunk 3]
```diff
@@ -430,6 +429,9 @@
             "drupal/node_revision_delete": {
                 "3249493 - Not able to delete a revision via the interface": "https://www.drupal.org/files/issues/2022-02-21/3249493-9.patch"
             },
+            "drupal/node_link_report": {
+                "3308887 - PHP 8.1 Deprecated Function": "https://www.drupal.org/files/issues/2022-09-09/Fix_for_php_8_1_deprecation.patch"
```
Comment (swirtSJW) [inline composer.json:433]: No need for this, This release will get picked up by the next dependabot bump https://www.drupal.org/project/node_link_report/releases/8.x-1.18

File: composer.json [hunk 4]
```diff
@@ -462,6 +464,9 @@
             "drupal/social_media_links": {
                 "3044002 - Platform Name and Aria Label issue": "https://www.drupal.org/files/issues/2022-03-21/3044002-17.patch"
             },
+            "drupal/styleguide": {
+                "3258816 - Fix [PHP 8.1] Deprecated function: strpos(): Passing null to parameter #1": "https://git.drupalcode.org/project/styleguide/-/merge_requests/5.diff"
```
Comment (swirtSJW) [inline composer.json:468]: We don't want to track the diff of a PR,  that will float with the PR.
https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/READMES/workflow.md#:~:text=If%20the%20patch%20is%20from%20a%20PR%20rather%20than%20an%20actual%20file%2C%20add%20it%20to%20the%20upstream%20issue%20as%20a%20file%20so%20that%20we%20can%20reference%20the%20file%20rather%20than%20a%20patch%20that%20will%20float%20with%20the%20upstream%20PR.%20A%20floating%20patch%20is%20a%20security%20risk.

**Review by swirtSJW — DISMISSED** [review]
Overall, great work on this.  A couple of issues of concern.

File: composer.json (hunk 3 above)
Comment (edmund-dunn) [reply composer.json:433]: Done.

File: composer.json (hunk 4 above)
Comment (edmund-dunn) [reply composer.json]: I fixed this following your model.

File: composer.json (hunk 2 above)
Comment (edmund-dunn) [reply composer.json]: Fixed this too. Updating "caxy/php-htmldiff" on its own worked.

**Comment by edmund-dunn** [comment]
@swirtSJW @ndouglas I downgraded drush back to 10.6 so we can push this live on PHP 7.4. Once we upgrade PHP to 8.1 we can update drush to 11.1.

**Review by ndouglas — COMMENTED** [review]
This looks great!

**Review by ndouglas — DISMISSED** [review]

**Review by ndouglas — DISMISSED** [review]

**Review by swirtSJW — DISMISSED** [review]
Looks like it all makes sense to me.  Nice job paving the way for an update.

**Pull request merged**